### PR TITLE
Added server/db and tests

### PR DESCRIPTION
**Pull request by jennqiao**
Server and db up and running with tests + Travis YML.

Please ignore the client side files - still working on those! 

Please review @2000prcs @sujithavikram @leealbert95

### CODE REVIEW

File: server/specs/db.test.js [hunk 1]
```diff
@@ -0,0 +1,119 @@
+const db = require('../db.js');
+
+
+describe('DB', () => {
+  
+  test('Should return an array when asked for similar listings', (done) => {
+    db.getSimilarListings(1, (err, result) => {
+      expect(Array.isArray(result)).toBe(true);
+      done();
```
Comment (sujithavikram) [inline server/specs/db.test.js]: @jennqiao can I know how do you get the keywords for similar listings

File: server/index.js [hunk 2]
```diff
@@ -0,0 +1,50 @@
+const express = require('express');
+let parser = require('body-parser');
+let morgan = require('morgan'); 
+let db = require('./db.js');
+
+let app = express();
+
+app.use(morgan('dev'));
```
Comment (sujithavikram) [inline server/index.js]: are you using morgan here?

File: server/specs/db.test.js [hunk 3]
```diff
@@ -0,0 +1,119 @@
...
+      } else {
+        expect(result[0].keywords.includes('home')).toBe(true);
+        done();
+      }
+     
+    });
+  });
+
+});
+
```
Comment (sujithavikram) [inline server/specs/db.test.js]: @jennqiao Everything looks good for me.
I have a doubt here, what is the result if there is no keywords matching? how will SimilarListings be displayed?

File: server/specs/db.test.js (hunk 1 above)
Comment (jennqiao) [reply server/specs/db.test.js]: Yup! You can sort of see it in the next db test. I take the keywords from the listing with the same ID, and then find other listings that have those keywords listed in their own keywords array.

File: server/index.js (hunk 2 above)
Comment (jennqiao) [reply server/index.js]: yup! I'm using it to log all the post/gets pings to my server.

File: server/specs/db.test.js (hunk 3 above)
Comment (jennqiao) [reply server/specs/db.test.js]: That's a great question. If there are no keywords matching (fewer than 3), it looks for price similarities (anything within $100 range of the price) and displays those instead.

**Pull request merged**